tests/test_acceptparse.py

```diff
@@ -466,1570 +475,426 @@ def test_parse_offer__invalid(self, offer):
-            (", ",),
-            {", ": 1.0},
-            {", ;level=1": (1.0, ";e1=1")},
-            "a/b, c/d;q=1;e1;",
-            ["a/b", "c/d;q=1;e1;"],
-            ("a/b", "c/d;q=1;e1;"),
-            {"a/b": 1.0, "cd": 1.0},
-            {"a/b": (1.0, ";e1=1"), "c/d": (1.0, ";e2=2;")},
+            ("", "<Accept('')>"),
+            (
+                r',,text/html ; p1="\"\1\"" ; q=0.50; e1=1 ;e2  ,  text/plain ,',
+                r"""<Accept('text/html;p1="\\"1\\"";q=0.5;e1=1;e2"""
+                + ", text/plain')>",
+            ),
+            (
+                ',\t, a/b ;  p1=1 ; p2=2 ;\t q=0.20 ;\te1="\\"\\1\\""\t; e2 ; '
+                + "e3=3, c/d ,,",
+                r"""<Accept('a/b;p1=1;p2=2;q=0.2;e1="\\"1\\"";e2""" + ";e3=3, c/d')>",
+            ),
         ],
     )
-    def test___add___invalid_value(self, right_operand):
-        left_operand = AcceptValidHeader(header_value="text/html")
-        result = left_operand + right_operand
-        assert isinstance(result, AcceptValidHeader)
-        assert result.header_value == left_operand.header_value
-        assert result is not left_operand
-
-    @pytest.mark.parametrize("str_", [", ", "a/b, c/d;q=1;e1;"])
-    def test___add___other_type_with_invalid___str__(self, str_):
-        left_operand = AcceptValidHeader(header_value="text/html")
-
-        class Other:
-            def __str__(self):
-                return str_
+    def test___repr__(self, header_value, expected_returned):
+        instance = Accept(header_value)
+        assert repr(instance) == expected_returned
 
-        right_operand = Other()
-        result = left_operand + right_operand
-        assert isinstance(result, AcceptValidHeader)
-        assert result.header_value == left_operand.header_value
-        assert result is not left_operand
+    @pytest.mark.parametrize(
+        "header_value, expected_returned",
+        [
+            ("", ""),
+            (
+                r',,text/html ; p1="\"\1\"" ; q=0.50; e1=1 ;e2  ,  text/plain ,',
+                r'text/html;p1="\"1\"";q=0.5;e1=1;e2, text/plain',
+            ),
+            (
+                ',\t, a/b ;  p1=1 ; p2=2 ;\t q=0.20 ;\te1="\\"\\1\\""\t; e2 ; '
+                + "e3=3, c/d ,,",
+                'a/b;p1=1;p2=2;q=0.2;e1="\\"1\\"";e2;e3=3, c/d',
+            ),
+        ],
+    )
+    def test___str__(self, header_value, expected_returned):
+        instance = Accept(header_value)
+        assert str(instance) == expected_returned
 
-    @pytest.mark.parametrize("value", ["", [], (), {}])
-    def test___add___valid_empty_value(self, value):
-        left_operand = AcceptValidHeader(header_value=",\t ,i/j, k/l;q=0.333,")
-        result = left_operand + value
-        assert isinstance(result, AcceptValidHeader)
-        assert result.header_value == left_operand.header_value
-        assert result is not left_operand
+    def test_copy(self):
+        instance = Accept("*/plain;charset=utf8;x-version=1")
```

Comment:
`*/plain` is valid according to https://datatracker.ietf.org/doc/html/rfc7231.html#section-5.3.2, but I think this is an oversight in the spec - I don't think they intended for it to be possible for `type` to be `*` when the `subtype` is not `*`. Not 100% certain about this, but as I think we can use any valid header value here, probably better to use one that we know is intended to be valid.

From the RFC: 'The "q" parameter is necessary if any extensions (accept-ext) are present, since it acts as a separator between the two parameter sets.' So this needs a "q" parameter before the extension parameter for the header to be valid (assuming that you want to use a valid header value here, and not an intentionally invalid one).

The missing "q" parameter does not seem like it's being detected by the regex at the moment - however, [extension parameters have been removed in RFC 9110](https://datatracker.ietf.org/doc/html/rfc9110#name-changes-from-rfc-7231), which obsoletes RFC 7231, so whether it's worth fixing may depend on when you would like WebOb to start conforming to RFC 9110?

Reply:
I'm open to removing extension param support - I don't have a strong opinion here. I'm not sure it's really hurting things much to keep them though.

Reply:
RFC 9110 says that "The accept extension grammar (accept-params, accept-ext) has been removed because it had a complicated definition, was not being used in practice, and is more easily deployed through new header fields." I imagine they must be pretty certain that it's not being used in practice, to feel able to make the change.

I don't see how it would hurt things to keep the extension param support either, but it'd be nice to simplify things and be able to remove some code, I imagine. Not sure what you think about moving to conform to RFC 9110 in general - as RFC 7231 is obsoleted by RFC 9110, I imagine WebOb would want to make the move at some point?

If you would like to keep the extension param support for the time being, I can look into why the regex did not catch the missing "q" parameter separator?